lib/puppet_x/templates/init_ps.ps1.erb

```diff
@@ -392,6 +396,12 @@ function Invoke-PowerShellUserCode
     [Void]$ps.AddScript($global:ourFunctions)
     $ps.Invoke()
 
+    if ([string]::IsNullOrEmpty($WorkingDirectory)) {
+      $ps.Runspace.SessionStateProxy.Path.SetLocation($global:DefaultWorkingDirectory)    
+    } else {
+      if (-not (Test-Path -Path $WorkingDirectory)) { Throw "Working directory `"$WorkingDirectory`" does not exist" }
+      $ps.Runspace.SessionStateProxy.Path.SetLocation($WorkingDirectory)
```

Comment:
One question - is there any situation where we wouldn't be able to `Se-tLocation` due to permissions but it would still be a valid path according to `Test-Path` ?  E.g. akin to on *nix without execute bit set? My quick testing showed no - always able to enter the most locked down folders, but I'm not sure.